bodhi/models/models.py

```diff
@@ -1662,13 +1677,18 @@ def check_karma_thresholds(self, db, agent):
         if not self.locked:
             if self.status is UpdateStatus.testing:
                 if self.stable_karma not in (0, None) and self.karma >= self.stable_karma:
-                    log.info("Automatically marking %s as stable" % self.title)
-                    self.set_request(db, UpdateRequest.stable, agent)
-                    self.request = UpdateRequest.stable
-                    self.date_pushed = None
-                    notifications.publish(
-                        topic='update.karma.threshold.reach',
-                        msg=dict(update=self, status='stable'))
+                    if self.autokarma:
+                        log.info("Automatically marking %s as stable" % self.title)
+                        self.set_request(db, UpdateRequest.stable, agent)
+                        self.request = UpdateRequest.stable
+                        self.date_pushed = None
+                        notifications.publish(
+                            topic='update.karma.threshold.reach',
+                            msg=dict(update=self, status='stable'))
+                    else:
+                        log.info("%s now reaches stable karma threshold and can be manually pushed to stable now" % self.title)
+                        self.request = None
+                        # We have to add the 'testing_approval_msg_based_on_karma' message as comment after this
```

Comment:
I don't think you necessarily need to delete this part... It's there for when an update reaches the karma threshold mid-push, then immediately after the push is done it gets the comment. The `scripts/approve-testing` cron job doesn't run as often, but it will still catch these updates... not a big deal either way.